plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileFormats.java

```diff
@@ -384,6 +386,24 @@ public void testIon(int rowCount, long fileSizePadding)
                 .isReadableByPageSource(fileSystemFactory -> new IonPageSourceFactory(fileSystemFactory));
     }
 
+    @Test(dataProvider = "validRowAndFileSizePadding")
+    public void testIonWithTextEncoding(int rowCount, long fileSizePadding)
```

Comment:
Having this test doesn't hurt, but I think we need a test (in `IonPageSourceFactorySmokeTest` or a new derivative) that proves text writes, well text, and binary writes binary.